### PR TITLE
cli-enhancement

### PR DESCRIPTION
**Pull request by aananthraj**
This PR contains the following code changes
- restructuring CLI commands and subcommands
  - `resource add` and `resource remove` subcommands added: to manage resources individually
  - `config view` subcommand added: to view the entire config file
  - help subcommand disabled and its access is limited to flags "-h" and "--help"
  - `version` subcommand fixed

- README.md updated: changes made to enhance usage and improve readability.
  rendered version on [README.md](https://github.com/aananthraj/kubewatch/tree/feature/cli-enhancement) is available here.

suggestions and comments ..?

### CODE REVIEW

**Comment by aananthraj** [comment]
@tylerauerbeck  can you help here..?

hope we could restart the process from now on and complete some missing thoughts form #99

**Comment by arapulido** [comment]
@jbianquetti-nami @jjo maybe you can have a look to this pr?

**Comment by aananthraj** [comment]
@arapulido, @jbianquetti-nami and @jjo  thanks for the prompt responce 👍.
Hope things go well.

File: README.md [hunk 1]
```diff
@@ -2,22 +2,50 @@
 
 [![Build Status](https://travis-ci.org/bitnami-labs/kubewatch.svg?branch=master)](https://travis-ci.org/bitnami-labs/kubewatch) [![Join us on Slack](https://s3.eu-central-1.amazonaws.com/ngtuna/join-us-on-slack.png)](https://skippbox.herokuapp.com)
 
-`kubewatch` is a Kubernetes watcher that currently publishes notification to Slack. Run it in your k8s cluster, and you will get event notifications in a slack channel.
+**kubewatch** is a Kubernetes watcher that currently publishes notification to Slack. Run it in your k8s cluster, and you will get event notifications in a slack channel.
```
[the remark this answers is not in the document]
Comment (aananthraj) [reply README.md]: `to slack` is now replaced with `to available collaboration hubs/notification channels.`

File: README.md [hunk 2]
```diff
@@ -45,7 +73,7 @@ And use that:
 $ helm upgrade --install kubewatch stable/kubewatch --values=values-file.yml
 ```
 
-## Installing kubewatch using kubectl
+#### Using kubectl:
 
 In order to run kubewatch in a Kubernetes cluster quickly, the easiest way is for you to create a [ConfigMap](https://github.com/bitnami-labs/kubewatch/blob/master/kubewatch-configmap.yaml) to hold kubewatch configuration. It contains the SLACK bot API token and channel to use.
```
[the remark this answers is not in the document]
Comment (aananthraj) [reply README.md]: removed them.

File: README.md [hunk 3]
```diff
@@ -110,81 +138,222 @@ Then just create `pod` as usual with:
...
-## Download kubewatch package
+### slack:
+
+- Create a [slack Bot](https://my.slack.com/services/new/bot)
+
+- Edit the Bot to customize its name, icon and retrieve the API token (it starts with `xoxb-`).
+
+- Invite the Bot into your channel by typing: `/join @name_of_your_bot` in the Slack message area.
+
+- Add Api token to kubewatch config using the follwoing steps
```
[the remark this answers is not in the document]
Comment (aananthraj) [reply README.md]: sry for this.
typo fixed :+1:

**Comment by aananthraj** [comment]
can this be merged?

**Comment by jjo** [comment]
> can this be merged?

Give me today/morrow to review it, thanks.

File: README.md [hunk 4]
```diff
@@ -110,81 +138,222 @@ Then just create `pod` as usual with:
...
-$ go get -u github.com/bitnami-labs/kubewatch
 ```
 
-## Configuration
 
-Kubewatch supports `config` command for configuration. Config file will be saved at $HOME/.kubewatch.yaml
+## Resources
 
-### Configure slack
+To manage the resources being watched, use the following command.
```
Comment (jjo) [inline README.md]: Suggest a small addition (besides being mentioned :

... use the following command, changes will be saved to `$HOME/.kubewatch.yaml`.

File: README.md [hunk 5]
```diff
@@ -110,81 +138,222 @@ Then just create `pod` as usual with:
...
+  $ kubewatch config add flock --url <flock_webhook_url>
+  ```
+  You have an altenative choice to set your FLOCK URL
+
+  ```console
+  $ export KW_FLOCK_URL='https://api.flock.com/hooks/sendMessage/XXXXXXXX'
+  ```
+
+## Viewing config
+To view the entire config file `.kubewatch.yaml` use the following command.
```
Comment (jjo) [inline README.md]: Please use full path `$HOME/.kubewatch.yaml`

File: cmd/resource.go [hunk 6]
```diff
@@ -24,9 +24,156 @@ import (
...
+		if err == nil {
+			if b == true {
+				conf.Resource.Services = b
+				logrus.Info("resource svc configured")
+			}
+		} else {
+			logrus.Fatal("svc", err)
+		}
+
+		b, err = cmd.Flags().GetBool("deployments")
```
Comment (jjo) [inline cmd/resource.go]: `deployments` is plural while others are used as singular (e.g. `svc`), please
use singular, and let's use the same Kubernetes abbreviated form, i.e.
`deploy`.

Ditto below with other cases of this.

File: cmd/resource.go [hunk 7]
```diff
@@ -24,9 +24,156 @@ import (
...
+		if err == nil {
+			if b == true {
+				conf.Resource.Namespace = b
+				logrus.Info("resource ns configured")
+			}
+		} else {
+			logrus.Fatal("ns", err)
+		}
+
+		b, err = cmd.Flags().GetBool("jobs")
```
Comment (jjo) [inline cmd/resource.go]: Use `job` (singular)

File: cmd/resource.go [hunk 8]
```diff
@@ -24,9 +24,156 @@ import (
...
+		if err == nil {
+			if b == true {
+				conf.Resource.Secret = b
+				logrus.Info("resource secret configured")
+			}
+		} else {
+			logrus.Fatal("secret", err)
+		}
+
+		b, err = cmd.Flags().GetBool("configmap")
```
Comment (jjo) [inline cmd/resource.go]: Use `cm` (same Kubernetes abbreviation)

File: README.md (hunk 4 above)
Comment (aananthraj) [reply README.md]: updated

File: README.md (hunk 5 above)
Comment (aananthraj) [reply README.md]: resolved

File: cmd/resource.go (hunk 7 above)
Comment (aananthraj) [reply cmd/resource.go]: changes done

File: cmd/resource.go (hunk 8 above)
Comment (aananthraj) [reply cmd/resource.go]: changes done

File: cmd/resource.go (hunk 6 above)
Comment (aananthraj) [reply cmd/resource.go]: changes done

**Comment by aananthraj** [comment]
@jjo thanks for the requested changes.
 
I have made those fixes and optimized `resource.go`. kindly have a look :)

**Comment by aananthraj** [comment]
@jjo any updates..? Any further changes required..?

**Comment by jjo** [comment]
Thanks @aananthraj for the code reorg, LGTM after final suggestion for clearer variable names and an other minors, PTAL at proposed https://github.com/aananthraj/kubewatch/pull/2 to merge into this PR.

**Comment by aananthraj** [comment]
Thanks @jjo for excellent support. :+1: , it is really nice to work with you peopel. :sunglasses: 
I have merged it. Hope its ready now. :100:

**Comment by jjo** [comment]
Thanks @aananthraj for the followup fixes, LGTM but let's tests pass,
I see that we're using `tip` as go version in `.travis.yml` which IMO doesn't
make much sense for this project, I'd rather try following latest go.

@aananthraj can you please apply below, to fix go version to `1.12.x` ?
```
$ git diff
diff --git a/.travis.yml b/.travis.yml
index effe08c..29cd6d0 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -3,8 +3,7 @@ language: go
 sudo: required
 
 go:
-  - 1.8
-  - tip
+  - 1.12.x
 
 branches:
```
@jbianquetti-nami would you agree with -^ ?

**Comment by jjo** [comment]
LGTM, thanks @aananthraj for fixing travis-CI !

**Comment by jjo** [comment]
@jbianquetti-nami fine to merge ? (your current status here is `requested changes`)

**Comment by aananthraj** [comment]
@jjo, just squashed all commit into a single one so that branches can be clean.
@jbianquetti-nami ,@jbianquetti kindly update your comments regarding the requested changes.
( guess we have github accounts mismatch here 😅)

**Comment by jbianquetti-nami** [comment]
> @jbianquetti-nami ,@jbianquetti
> ( guess we have github accounts mismatch here 😅)

No, double hit. Approving.

**Comment by aananthraj** [comment]
Feeling really happy for this :man_dancing: . 
Thanks @arapulido, @jbianquetti-nami and @jjo for the extended support on this PR.
Hope we could improve Kubewatch by working togeather :+1: .

**Pull request merged**